driver/ixsystems/common.py

```diff
@@ -339,7 +339,14 @@ def _delete_volume(self, name):
             fullvolume, snapname = clone.split('@')
             temp, snapvol = fullvolume.rsplit('/', 1)
             self._delete_snapshot(snapname, snapvol)
-        if ret['status'] != FreeNASServer.STATUS_OK:
+
+        # When deleting volume with dependent snapsnot clone, 422 error triggered. Throw VolumeIsBus exception ensures
+        # upper stream cinder manager mark volume status available instead of error-deleting.
+        if ret['status'] == 'error' and ret['response'] == '422:Unprocessable Entity':
```

Comment:
That seems like a weird check, don't we get only the status code (422) from the API response?

Reply:
There is additional message from api response "Failed to delete dataset: cannot destroy 'pool/cinder/volume-21030469': volume has children
use '-r' to destroy the following datasets:
pool/cinder/volume-21030469@snap-9f689fb8" from response message field. My understand this message content changes, so I use http response code and http response message "422:Unprocessable Entity". I dont know how to use this additional message information yet.

Reply:
HTTP should return only the `422` which seems to be more reliable than the string "Unprocessable Entity" which may vary depending on the http server implementation

Reply:
Thank you!
The current ix driver error handling internal dict, compress the response code with response msg, 
https://github.com/iXsystems/cinder/blob/master/driver/ixsystems/freenasapi.py#L183            self.COMMAND_RESPONSE['response'] = '%d:%s' % (err.code, err.msg),

Let me enhance this by put error code as separate field, and use http error code to throw deviceisbusy exeception so that it wont depends on http server underlying implementation message like you mentioned:"Unprocessable Entity".